### PR TITLE
OD-581 [Feature] added translation to Widget-Lock

### PR DESCRIPTION
**Pull request by OleksandrLavrynenko**
@sofiiakvasnevska @inna-bieshulia

OD-581 https://weboo.atlassian.net/browse/OD-581

## Description
Translations first try

## Backward compatibility
This change is fully backward compatible.

## Reviewers
@upplabs-alex-levchenko

### CODE REVIEW

File: build.html [hunk 1]
```diff
@@ -1,42 +1,42 @@
...
-    <p>To protect your data and unlock your app you will need to set up a 4-digit passcode</p>
-    <p class="text-danger">Passcodes did not match. Please try again.</p>
+    <h1 data-translate="widgets.lock.dataSource.setup.title">Set up a Passcode</h1>
+    <input type="tel" class="passcode-field num-setup" data-translate="[placeholder]widgets.lock.dataSource.setup.passwordPlaceholder" placeholder="Tap to enter passcode" />
+    <p data-translate="widgets.lock.dataSource.setup.instruction">To protect your data and unlock your app you will need to set up a 4-digit passcode</p>
+    <p class="text-danger" data-translate="widgets.lock.dataSource.setup.mismatch">Passcodes did not match. Please try again.</p>
   </div>
   <div class="state future" data-state="verify">
-    <h1>Verify your Passcode</h1>
+    <h1 data-translate="widgets.lock.dataSource.verify.title">verify</h1>
```
Comment (tonytlwu) [inline build.html]: Why has the H1 content changed from `Verify your Passcode` to `verify`?

File: js/build.js [hunk 2]
```diff
@@ -3,6 +3,10 @@ Fliplet.Widget.instance('lock', function(data) {
   var widgetId = data.id;
   var widgetUuid = data.uuid;
 
+  $(this).translate();
+  
+  console.log(T('widgets.login.dataSource.errors.sessionNotFound'));
```
Comment (tonytlwu) [inline js/build.js]: Why is this here?

File: build.html [hunk 3]
```diff
@@ -1,42 +1,42 @@
...
-    <p>Enter your passcode to unlock your app.</p>
-    <p><a href="#" class="btn btn-link forgot-passcode">Forgot your passcode?</a></p>
-    <p class="text-danger">Incorrect passcode. Please try again.</p>
-    <div class="use-touchid notShow">
+    <h1 data-translate="widgets.lock.dataSource.unlock.title">Unlock your app</h1>
+    <input type="tel" class="passcode-field num-unlock" data-translate="[placeholder]widgets.lock.dataSource.unlock.passwordPlaceholder" placeholder="Tap to enter passcode" />
+    <p data-translate="widgets.lock.dataSource.unlock.instruction">Enter your passcode to unlock your app.</p>
+    <p><a href="#" class="btn btn-link forgot-passcode" data-translate="widgets.lock.dataSource.unlock.forgotPassword">Forgot your passcode?</a></p>
+    <p class="text-danger" data-translate="widgets.lock.dataSource.unlock.mismatch">Incorrect passcode. Please try again.</p>
+    <div class="use-touchid notShow" data-translate="widgets.lock.dataSource.unlock.message">
```
Comment (tonytlwu) [inline build.html]: Because this translation string involves HTML too, can you verify, based on [this implementation](https://github.com/Fliplet/fliplet-widget-login-data-source/pull/96/files#diff-1e1cae91d18fbf32b276399bcdce1fc2380580c17323539e9457d0888d3b7b26R60), what difference it makes to implement it like you have done here vs using `[html]`?

**Comment by OleksandrLavrynenko** [comment]
@tonytlwu I made corrections and update pull request

**Review by tonytlwu — COMMENTED** [review]
This all looks good now. But we can't approve or merge this until the translation backend is ready.

File: translation.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{
+    "widgets": {
+        "lock": {
+            "dataSource": {
```
Comment (tonytlwu) [inline translation.json]:
```suggestion
```

File: translation.json [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+                    "passwordPlaceholder": "Tap to enter passcode",
+                    "instruction": "To protect your data and unlock your app you will need to set up a 4-digit passcode",
+                    "mismatch": "Passcodes did not match. Please try again."
+                },
+                "verify": {
+                    "title": "Verify your Passcode",
+                    "passwordPlaceholder": "Tap to re-enter passcode",
+                    "instruction": "To protect your data and unlock your app you will need to set up a 4-digit passcode"
+                },
+                "touchID": {
```
Comment (tonytlwu) [inline translation.json]:
```suggestion
                "touchId": {
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: translation.json [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                "passwordPlaceholder": "Tap to re-enter passcode",
+                "instruction": "To protect your data and unlock your app you will need to set up a 4-digit passcode"
+            },
+            "touchId": {
+                "title": "Complete",
+                "success": "Your passcode is successfully set up.",
+                "instruction": "You can also use <span class=\"bio-id-type\">Face ID</span> to unlock your app in the future.",
+                "action": "Continue"
+            },
+            "noTouchID": {
```
Comment (tonytlwu) [inline translation.json]:
```suggestion
            "noTouchId": {
```

File: build.html [hunk 7]
```diff
@@ -1,42 +1,42 @@
...
-    <p>To protect your data and unlock your app you will need to set up a 4-digit passcode</p>
+    <p data-translate="widgets.lock.verify.instruction">To protect your data and unlock your app you will need to set up a 4-digit passcode</p>
   </div>
   <div class="state future" data-state="touchID">
-    <h1>Complete</h1>
-    <p>Your passcode is successfully set up.</p>
-    <p>You can also use <span class="bio-id-type">Face ID</span> to unlock your app in the future.</p>
-    <button class="btn btn-primary continue-touchID">Continue</button>
+    <h1 data-translate="widgets.lock.touchId.title">Complete</h1>
+    <p data-translate="widgets.lock.touchID.success">Your passcode is successfully set up.</p>
```
Comment (tonytlwu) [inline build.html]: This shows the changes weren't tested after they are changed from `touchID` to `touchId`. Please show screencast of all the messages working.

```suggestion
    <p data-translate="widgets.lock.touchId.success">Your passcode is successfully set up.</p>
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: build.html (hunk 7 above)
Comment (OleksandrLavrynenko) [reply build.html]: > This shows the changes weren't tested after they are changed from `touchID` to `touchId`. Please show screencast of all the messages working.

Yes, I'm sorry. I missed it.

File: translation.json [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+      },
+      "unlock": {
+        "title": "Unlock your app",
+        "passwordPlaceholder": "Tap to enter passcode",
+        "instruction": "Enter your passcode to unlock your app.",
+        "forgotPassword": "Forgot your passcode?",
+        "mismatch": "Incorrect passcode. Please try again.",
+        "message": "Use <span class=\"bio-id-type\">Face ID</span>"
+      },
+      "errors": {
```
Comment (upplabs-alex-levchenko) [inline translation.json:42]: этот блок мне весь не нравится

File: translation.json [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        }
+      },
+      "noTouchId": {
+        "title": "Complete",
+        "success": "Your passcode is successfully set up.",
+        "actions": {
+          "continue": "Continue"
+        }
+      },
+      "unlock": {
```
Comment (upplabs-alex-levchenko) [inline translation.json:34]: mismatch это error какой-то?

**Review by tonytlwu — APPROVED** [review]
@romanyosyfiv @inna-bieshulia Translation looks ready. This can be tested once API is ready.
